packages/core/upload/admin/src/utils/appendSearchParamsToUrl.js

```diff
@@ -10,7 +10,7 @@ const appendSearchParamsToUrl = ({ url, params }) => {
     return url;
   }
 
-  const urlObj = new URL(url, window.strapi.backendURL);
+  const urlObj = new URL(url, window.strapi.backendURL ?? window.location.origin);
```

Comment:
Does it account for empty string ?

Reply:
Why would `window.strapi.backendURL` be an empty string? it's defined only by `process.env.STRAPI_ADMIN_BACKEND_URL` from what I can see. 🤔

Reply:
Seems it is in production (url error)... not with yarn develop (maybe related to how webpack replace them). I checked an hour ago with the latest version... 

If you replace by '||' it works... but don't know if it shouldn't be fixed somewhere else.

Reply:
I've proposed a patch here: https://github.com/strapi/strapi/pull/18022 That should fix it but up to review cause I'm not sure if it shouldn't be done more globally